scripts/us_eia/opendata/process/coal.py

```diff
@@ -0,0 +1,236 @@
+        measure = m.group(2)
+        if activity in ['EXPORT', 'IMPORT']:
+            # Pattern #2
+            # TODO: model destination / source port as well
+            type = m.group(3)
+            place = m.group(4)
+            port = m.group(5)
+            period = m.group(6)
+            sv_id = f'COAL.{activity}_{measure}.{type}.{period}'
+            return (place, sv_id, False)
+        elif activity == 'SHIPMENT':
+            # Pattern #3
+            source = m.group(3)
+            if source.isalpha():  # could include 3-letter region codes
+                destination_power_plant = m.group(4)
+                material = m.group(5)
+                period = m.group(6)
+                sv_id = f'COAL.SHIPMENT_{measure}.{material}.{period}'
+                return (source, sv_id, True)
+            else:
+                # TODO: Handle remaining places - coal mines
+                counters[f'error_unknown_coal_mine SHIPMENT '] += 1
+                return (None, None, None)
+        else:
+            counters[f'unknown #2,3 activity ({activity})'] += 1
+            return (None, None, None)
+
+    # Pattern #4: COAL.PROD_DIST_STOCKS.TOT-{Place}.{Period}
+    # Pattern #4: COAL.PRICE_BY_RANK.{Region}-{Material}.{Period}
+    # Pattern #4: COAL.SHIP_{MINE|PLANT}_{ASH|HEAT|PRICE|QTY|SULFUR}.{Region}-{Material}.{Period}
+    m = re.match(r"^COAL\.([A-Z]+_[A-Z]+_[A-Z]+)\.([^-]+)-([^.]+)\.([AQM])$",
+                 series_id)
+    if m:
+        measure = m.group(1)
+        if measure == "PROD_DIST_STOCKS":
+            assert m.group(2) == "TOT"
+            place = m.group(3)
+            period = m.group(4)
+            return (place, f'COAL.PROD_DIST_STOCKS.TOT.{period}', True)
+        else:
+            place = m.group(2)
+            material = m.group(3)
+            period = m.group(4)
+            return (place, f'COAL.{measure}.{material}.{period}', True)
+
+    return (None, None, None)
+
+
+##
+## Maps for Schema - more definitions at https://www.eia.gov/coal/data/browser/data/termsAndDefs.php?rseAvailable=false&showFilterValues=true&showDetail=true&showTransportationMode=true&showPrimeMovers=true&showPlantFuelTypes=true&showMineType=true&showMineStatus=true&topic=26
+##
+
+# Each value is a list where first entry is StatVar ID component, and the rest
+# are StatVar PVs.
+_MEASURE_MAP = {
+    'ASH_CONTENT': [
+        'AverageQuality_AshContent_ReceivedCoal',
+        'populationType: dcs:Coal',
+        'measuredProperty: dcs:ashContent',
+        'statType: dcs:meanValue',
+    ],
+    'CONS_TOT': [
+        'Consumption_Coal',
+        'populationType: dcs:Coal',
+        'measuredProperty: dcs:consumption',
+        'statType: dcs:measuredValue',
+    ],
+    'COST': [
+        'Average_Cost_Coal',
+        'populationType: dcs:Coal',
+        'measuredProperty: dcs:cost',
+        'statType: dcs:meanValue',
+    ],
+    'HEAT_CONTENT': [
+        'AverageQuality_HeatContent_ReceivedCoal',
```

Comment:
Just "Average" here?

(I thought AverageQuality was used for ash content since it talks about quality of coal)